packages/rx_bloc_cli/mason_templates/bricks/feature_onboarding/__brick__/lib/feature_onboarding/blocs/onboarding_bloc.dart

```diff
@@ -0,0 +1,206 @@
+  Stream<bool> get showFieldErrors;
+
+  /// The loading state
+  Stream<bool> get isLoading;
+
+  /// The error state
+  Stream<ErrorModel> get errors;
+
+  /// The error state for resuming the onboarding process
+  Stream<ErrorModel> get resumeOnboardingErrors;
+
+  /// The routing state
+  ConnectableStream<void> get onRouting;
+}
+
+@RxBloc()
+class OnboardingBloc extends $OnboardingBloc {
+  OnboardingBloc(
+    this._authService,
+    this._permissionsService,
+    this._onboardingService,
+    this._validatorService,
+    this._routerBloc,
+  ) {
+    registered.connect().addTo(_compositeSubscription);
+    onRouting.connect().addTo(_compositeSubscription);
+
+    resumeOnboardingErrors.listen((_) {});
+    _resumeOnboardingIfHasAuthToken();
+  }
+
+  Future<void> _resumeOnboardingIfHasAuthToken() async {
+    if (await _authService.isAuthenticated()) {
+      resumeOnboarding();
+    }
+  }
+
+  final AuthService _authService;
+  final PermissionsService _permissionsService;
+  final OnboardingService _onboardingService;
+  final LoginValidatorService _validatorService;
+  final RouterBlocType _routerBloc;
+
+  @override
+  Stream<String> _mapToEmailState() => _$setEmailEvent
+      .map(_validatorService.validateEmail)
+      .startWith('')
+      .shareReplay(maxSize: 1);
+
+  @override
+  Stream<String> _mapToPasswordState() => _$setPasswordEvent
+      .map(_validatorService.validatePassword)
+      .startWith('')
+      .shareReplay(maxSize: 1);
+
+  @override
+  ConnectableStream<UserWithAuthTokenModel> _mapToRegisteredState() => _$registerEvent
+      .throttleTime(const Duration(seconds: 1))
+      .withLatestFrom2<Result<String>, Result<String>, CredentialsModel?>(
+          email.asResultStream(),
+          password.asResultStream(),
+          (_, emailResult, passwordResult) => _validateAndReturnCredentials(
+                emailResult,
+                passwordResult,
+              ))
+      .where((args) => args != null)
+      .exhaustMap(
+        (args) => _onboardingService
+            .register(email: args!.email, password: args.password)
+            .asResultStream(),
+      )
+      .setResultStateHandler(this)
+      .whereSuccess()
+      .doOnData(
+          (userWithAuthToken) => navigateToNextStep(userWithAuthToken))
+      .publish();
+
+  CredentialsModel? _validateAndReturnCredentials(
+      Result<String> emailResult, Result<String> passwordResult) {
+    if (emailResult is ResultError || passwordResult is ResultError) {
+      return null;
+    }
+    if (emailResult is ResultLoading || passwordResult is ResultLoading) {
+      return null;
+    }
+
+    return CredentialsModel(
+      email: (emailResult as ResultSuccess<String>).data,
+      password: (passwordResult as ResultSuccess<String>).data.toSha256(),
+    );
+  }
+
+  Future<void> _saveToken(AuthTokenModel authToken) async {
+    await _onboardingService.setIsProfileTemporary(true);
```

Comment:
As discussed `setIsProfileTemporary` won't be needed

Reply:
It's been removed